### PR TITLE
Support for native bracket pair colorization

### PR DESCRIPTION
**Pull request by SpicyRicecaker**
With the release of VSCode 1.60, [native bracket pair colorization](https://code.visualstudio.com/updates/v1_60#_high-performance-bracket-pair-colorization) has been added, and can be enabled with the `"editor.bracketPairColorization.enabled": true` in settings. However, the default colors of the bracket pairs, as shown below, do not match very well with the gruvbox-material theme. 
image

So I basically made these changes
from the default vscode colors...
```json
"editorBracketHighlight.foreground1": "#ffd700",
"editorBracketHighlight.foreground2": "#da70d6",
"editorBracketHighlight.foreground3": "#179fff",
```
...to the default red, yellow, and blue colors from the gruvbox-material palette
```json
"editorBracketHighlight.foreground1": "#d8a657",
"editorBracketHighlight.foreground2": "#ea6962",
"editorBracketHighlight.foreground3": "#7daea3",
```

The result would make the same function look like this:
image

I'm totally down to further discuss and modify the 3+ colors for bracket colorization in this pull request; Honestly, I don't have any color theory knowledge at all, so I was thinking that'd I'd just put all the fields in the right place and get your take on it ;) 

Thanks a lot for making gruvbox-material btw, I've been using it for ~1 yr now and it's my favorite theme!

### CODE REVIEW

File: src/workbench/material.ts [hunk 1]
```diff
@@ -362,6 +362,9 @@ export function materialWorkbench(
     "editor.foldBackground": `${palette.bg5}80`,
     "editorBracketMatch.border": `${palette.bg}00`,
     "editorBracketMatch.background": `${palette.grey0}80`,
+    "editorBracketHighlight.foreground1": `${palette.yellow}`,
+    "editorBracketHighlight.foreground2": `${palette.red}`,
+    "editorBracketHighlight.foreground3": `${palette.blue}`,
```
Comment (sainnhe) [inline src/workbench/material.ts:367]: There are 6 available foreground tokens but you only added 3. Here are some suggestions:

```typescript
    "editorBracketHighlight.foreground1": `${palette.red}`,
    "editorBracketHighlight.foreground2": `${palette.yellow}`,
    "editorBracketHighlight.foreground3": `${palette.green}`,
    "editorBracketHighlight.foreground4": `${palette.blue}`,
    "editorBracketHighlight.foreground5": `${palette.orange}`,
    "editorBracketHighlight.foreground6": `${palette.purple}`,
    "editorBracketHighlight.unexpectedBracket.foreground": `${palette.grey1}`,
```

The key here is that you need to make the distinction between two adjacent parentheses large enough.

**Comment by sainnhe** [comment]
Another thing is that the new version number should be `6.4.6` since there is only a small fix/feature.

**Comment by SpicyRicecaker** [comment]
Thanks for the suggestions, I added the 6 `foreground` tokens as well as the `unexpectedBracket` tokens according to your comment, as well as bumped down the version down to `6.4.6`

Some quick questions: 
I was under the impression that the minor version of semantic versioning `6.4.x` was for bug fixes and no feature changes only? Does that mean that my understanding of semantic versioning is wrong, or does this project not follow it? 

I also did
```
git commit --amend
```
and 
```
git push origin master --force
```
to merge the would be two commits in this pull request into 1, was that the right move?

**Comment by sainnhe** [comment]
> I was under the impression that the minor version of semantic versioning 6.4.x was for bug fixes and no feature changes only?

Yes exactly, but this color scheme doesn't follow this convention, because there are almost no "bugs" in a color scheme, everything can be regarded as features.

So what I'm doing is:

- `6.4.x`: Small features with no configuration changes.
- `6.x.0`: When configuration options change.
- `x.0.0`: Break changes, often when some designs change.

**Comment by sainnhe** [comment]
Thanks for your contribution! I'm going to release a new version later.

**Comment by SpicyRicecaker** [comment]
np!
do you mind if I add what you said about versioning to contributing.md?

**Pull request merged**